### ISSUE TITLE
Enabling generateAsInternal in combination with nonempty sealedClassesForEnumsMatching generates invalid classes definitions

### ISSUE THREAD

**Issue by marosseleng** [apollographql/apollo-kotlin#6212]
### Version

4.0.1

### Summary

I have my apollo service defined in `build.gradle.kts` as follows:

```kotlin
apollo {
    service("myService") {
        packageName.set("...")
        schemaFile.set(file("..."))

        introspection {
            ...
        }
        generateAsInternal = true
        sealedClassesForEnumsMatching.set(setOf(".*"))
    }
}
```

I enabled `sealedClassesForEnumsMatching` just recently, because I wanted to access the raw value of the `UNKNOWN__` constant. However, the generated classes for `UNKNOWN__` constant looks like this:

```kotlin

private internal class UNKNOWN__MyModelClass(
  override val rawValue: String,
) : MyModelClass.UNKNOWN__ {
  override fun equals(other: Any?): Boolean {
    if (other !is UNKNOWN__MyModelClass) return false
    return this.rawValue == other.rawValue
  }

  override fun hashCode(): Int = this.rawValue.hashCode()

  override fun toString(): String = "UNKNOWN__($rawValue)"
}

```

If i disable `generateAsInternal = true` the problem doesn't occur, only `private` modifier remains on the generated class' definition.

The build then fails on invalid syntax.

I believe this is a bug.

**Comment by BoD** [comment]
Thanks for reporting! This was indeed a bug in the codegen - a fix is in #6213.

**Comment by BoD** [comment]
The fix will be in the next release. In the meantime you can test it with the [snapshots](https://www.apollographql.com/docs/kotlin#snapshots).

**Comment by martinbonnin** [comment]
The fix is now available in [4.1.0](https://github.com/apollographql/apollo-kotlin/releases/tag/v4.1.0)
